### PR TITLE
Create method to update account balance for custom accounts

### PR DESCRIPTION
**Pull request by CarterMcAlister**
Created a method that can be used to update the balance of custom accounts.
Account name in Personal Capital and new balance are accepted as parameters.

This can be used to write programs that scrape data from sources not supported by Personal Capital, and update the balance in Personal Capital.

### CODE REVIEW

File: personalcapital/personalcapital.js [hunk 1]
```diff
@@ -207,6 +208,38 @@ class PersonalCapital {
...
+
+    const selectedAccount = accounts.filter(account => {
+      return account.name === accountName;
+    });
+    if (!selectedAccount || selectedAccount.length === 0) {
+      throw `Account ${accountName} not found!`;
+    }
+
+    return selectedAccount[0];
+  }
```
Comment (JamaicanMoose) [inline personalcapital/personalcapital.js]: Make this more concise with something like this:
```js
async getAccountByName(accountName /*String*/) {
    const accounts = await this.getAccounts();
    const selectedAccount = accounts.find(account => account.name === accountName);
    if (!selectedAccount) {
      throw new AccountNotFoundError(`Account ${accountName} not found!`);
    }
    return selectedAccount;
  }
```

File: personalcapital/personalcapital.js [hunk 2]
```diff
@@ -207,6 +208,38 @@ class PersonalCapital {
...
+      ...account,
+      isTransferPending: false,
+      isTransferEligible: true,
+      ownershipType: null,
+      employerMatchLimitType: "dollar",
+      requestSource: "USER",
+      balance: newBalance,
+      currentBalance: newBalance,
+      availableBalance: newBalance
+    };
```
Comment (JamaicanMoose) [inline personalcapital/personalcapital.js]: Payloads should be defined in payload.endpoint as a function. Also ...account seems a bit extreme, does this actually require the whole account object or just the accountid?

**Review by JamaicanMoose — CHANGES_REQUESTED** [review]
Commit works and good catch on the password argument but needs some cleanup before merging.

File: personalcapital/personalcapital.js (hunk 1 above)
Comment (JamaicanMoose) [reply personalcapital/personalcapital.js]: Forgot, AccountNotFoundError should be defined on line 35 as `class AccountNotFoundError extends Error {}` and exported in the module.exports.

**Comment by JamaicanMoose** [comment]
Password argument change integrated in patch version 2.0.1.

**Comment by CarterMcAlister** [comment]
@JamaicanMoose Thanks for the feedback, I have made your requested changes. 
Inspecting the `updatebalance` call in the network tab shows that the PC website makes the call by including the entire account object. I did try with only the `accountid`, but it does not update the balance.

**Comment by JamaicanMoose** [comment]
@CarterMcAlister i did some inspection myself and confirmed that the website does use many of the parameters from the account object when updating some manual account types. I also noted that updating OTHER_ASSETS requires very few (however your code works when updating it presumably because the server ignores it). If you remove ownershipType: null (which as far as I can tell would override any existing ownershipType in the account object and make a note in the readme and function comment noting the asset type the payload was designed for (and that the code might work for other account types) I'm all good with approving it.

**Comment by CarterMcAlister** [comment]
@JamaicanMoose I have removed ownership types and added clarifying comments as to what asset types are supported by `updateBalance`. I have also added methods to support the `Manual Investment Holdings` asset.

**Review by JamaicanMoose — APPROVED** [review]

**Pull request merged**